rust/src/router/data_consumer.rs

```diff
@@ -509,24 +509,23 @@ impl DataConsumer {
                         }
                         Notification::DataProducerPause => {
                             let mut data_producer_paused = data_producer_paused.lock();
-                            let was_paused = *paused.lock() || *data_producer_paused;
+                            let paused = *paused.lock();
                             *data_producer_paused = true;
 
                             handlers.data_producer_pause.call_simple();
 
-                            if !was_paused {
+                            if !paused {
```

Comment:
No need to check `*data_producer_paused`, anymore?

Reply:
If worker sent "data producer pause" notification it means that data producer was not paused. No need to check that since it cannot be changed any other way.